File: Springboot_API/RetaCantabria/src/main/java/org/example/DTO/ValoracionDTO.java
package org.example.DTO;

import org.example.Entidades.Valoracion;

import java.time.LocalDateTime;
import java.time.OffsetDateTime;
import java.time.ZoneOffset;

public record ValoracionDTO(
        Long idRuta,
        Long idUsuario,
        Integer dificultad,
        Integer belleza,
        Integer interesCultural,
        OffsetDateTime fecha
) {
    public ValoracionDTO(Valoracion val) {
        this(
                val.getRuta().getIdRuta(),
                val.getUsuario().getIdUsuario(),
                val.getDificultad(),
                val.getBelleza(),
                val.getInteresCultural(),
                OffsetDateTime.of(val.getFecha(), ZoneOffset.of("-08:00"))
        );
    }
}

File: Aplicación Escritorio/ModeloTDO/valoracionDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ModeloDTO
{
    public class valoracionDTO
    {
            public long idRuta { get; set; }
            public long idUsuario { get; set; }
            public int dificultad { get; set; }
        public int belleza { get; set; }
        public int interesCultural { get; set; }
        public DateTimeOffset fecha { get; set; }
        }
    }
